src/main/java/bitbot/Deadline.java

```diff
@@ -0,0 +1,82 @@
+package bitbot;
+
+import java.time.LocalDate;
+import java.time.LocalDateTime;
+import java.time.LocalTime;
+import java.time.format.DateTimeFormatter;
+/**
+ * Adapted from the partial solution in the question
+ */
+public class Deadline extends Task {
+    private DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+    private DateTimeFormatter dateFormatter = DateTimeFormatter.ofPattern("MMM dd yyyy");
+    private DateTimeFormatter timeFormatter = DateTimeFormatter.ofPattern("HH:mm");
+
+    protected String by;
+    protected LocalDateTime localDateTime;
+    protected LocalDate localDate;
+    protected LocalTime localTime;
+
+    public Deadline (String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    public Deadline (String description, LocalDateTime localDateTime) {
+        super(description);
+        this.localDateTime = localDateTime;
+    }
+
+    public Deadline (String description, LocalDate localDate) {
+        super(description);
+        this.localDate = localDate;
+    }
+
+    public Deadline (String description, LocalTime localTime) {
+        super(description);
+        this.localTime = localTime;
+    }
+
+    /**
+     * prints out the final display of the task details to the user.
```

Comment:
Consider capitalizing the first letter of the opening sentence of each header comment for consistency. Same for description of return.